src/pynxtools_xps/kratos/metadata_kratos.py

```diff
@@ -0,0 +1,350 @@
+                flattened_dict[key] = value
+
+        for key in flattened_dict.copy().keys():
+            if "_units" in key:
+                new_key = key.replace("_units", "/@units")
+                flattened_dict[new_key] = flattened_dict.pop(key)
+
+        for key in flattened_dict.copy().keys():
+            if key in KEYS_WITH_UNITS and f"{key}/@units" not in flattened_dict:
+                flattened_dict[f"{key}/@units"] = UNIT_MISSING[key]
+
+        return flattened_dict
+
+
+def _parse_datetime(datetime_string: str) -> str:
+    """
+    Convert the native time format to the datetime string
+    in the ISO 8601 format: '%Y-%b-%dT%H:%M:%S.%fZ'.
+
+    Parameters
+    ----------
+    value : str
+        String representation of the date in the format
+        "%Y-%m-%d", "%m/%d/%Y" or "%H:%M:%S", "%I:%M:%S %p".
+
+    Returns
+    -------
+    date_object : str
+        Datetime in ISO8601 format.
+
+    """
+    possible_date_formats = ["%d.%m.%Y %H:%M", "%d/%m/%Y %H:%M"]
+    for date_fmt in possible_date_formats:
+        try:
+            datetime_obj = datetime.datetime.strptime(datetime_string, date_fmt)
+            return datetime_obj.astimezone().isoformat()
+
+        except ValueError:
+            continue
+    raise ValueError("Date and time could not be converted to ISO 8601 format.")
+
+
+def _convert_description(value: str):
+    """Map all items in description to a dictionary."""
+    pattern = re.compile(
+        r"\(\s*([-+]?\d*\.?\d+)\s*,\s*([-+]?\d*\.?\d+)\s*,\s*([-+]?\d*\.?\d+)\s*\)\s*([a-zA-Z]+)"
+    )
+    match = pattern.match(value)
+
+    if match:
+        x, y, z, unit = match.groups()
+        return {
+            "x": x,
+            "x_units": unit,
+            "y": y,
+            "y_units": unit,
+            "z": z,
+            "z_units": unit,
+        }
+    else:
+        raise ValueError(f"Invalid input string '{value}' for description.")
+
+
+def _separate_val_and_unit(value: str):
+    """Map all items in energy_referencing to a dictionary."""
+
+    pattern = re.compile(r"([-+]?\d*\.?\d+(?:[eE][-+]?\d+)?)([a-zA-Z]+)")
+    match = pattern.match(value)
+
+    if match:
+        value, unit = match.groups()
+        return value, unit
+    else:
+        raise ValueError(f"Input string '{value}' does not contain a value and unit.")
+
+
+def _convert_xray_deflection(value: str):
+    """Convert deflection like (0.000, 0.000)mm to dict."""
+
+    pattern = re.compile(
+        r"\(\s*([-+]?\d*\.?\d+)\s*,\s*([-+]?\d*\.?\d+)\s*\)\s*([a-zA-Z]+)"
+    )
+    match = pattern.match(value)
+
+    if match:
+        x, y, unit = match.groups()
+        return {
+            "x": float(x),
+            "y": float(y),
+            "x_units": unit,
+            "y_units": unit,
+        }
+    else:
+        raise ValueError(f"Invalid input string: '{value}' for X-ray deflection.")
+
+
+def _convert_spectrum_type(key, value: str):
+    return value
+    # wide : 286.690-1491.690eV, Sweeps 3
```

Comment:
This function is not doing anything. Right! you can remove it.

Reply:
I thought about doing something in the future, but probably won't happen. Removed.